Source/ARTTypes.m

```diff
@@ -387,20 +387,25 @@ - (nullable NSData *)art_archive {
 #endif
 }
 
-+ (nullable id)art_unarchive:(NSData *)data {
++ (nullable id)art_unarchiveWithAllowedClass:(Class)cls fromData:(NSData *)data {
```

Comment:
I'm a bit confused about why the `cls` parameter is now required. Why was it no longer sufficient to use `[self class]` as the expected class (in addition to the `NSArray` and `NSDictionary` that you've added)?

Reply:
Xcode now gives a warning when you use functions without `Class` parameter. See
`+ (nullable id)unarchiveObjectWithData:(NSData *)data API_DEPRECATED`

Reply:
@lawrence-forooghian

Reply:
<img width="1440" alt="Screenshot 2021-10-29 at 23 41 00" src="https://user-images.githubusercontent.com/45491869/139504789-b90ba287-eff1-4beb-8361-a474f0e58af0.png">

Reply:
I suspect that the intention of the author of this `art_unarchive` method (i.e. what I think that `[self class]` was trying to achieve) was that it would be called directly on the class that you wish to unarchive. For example, if you want to unarchive a `ARTPushActivationState`, you'd call `[ARTPushActivationState art_unarchive:data]`. If we were to change the call sites of the `art_unarchive` to call the method on the class instead of on `NSObject` then I don't think we'd need the `cls` parameter.

Reply:
It was true prior Xcode 13. v13 keeps spitting complaining warnings about that.

Reply:
Actually you are right, `[self class]` is pretty enough. I have no idea why I've refused to use <class> instead of `NSObject`, because "the author of this art_unarchive method" is me 😅 Probably because I played around with this method and then forgotten to replace `NSObject` to the actual classes or something like that.

Reply:
Fixed e934771